### DISCUSSION TITLE
utilizzo "eleapi"

### DISCUSSION THREAD

**Discussion by giuspataro** [ondata/elezioni-politiche-2022#25] [Domande e risposte]
Salve a tutti. Intanto complimenti per il lavoro stupendo che fate. Detto ciò, mi sono ritrovato a dover scaricare anche io i dati delle elezioni prima che arrivassero i vostri, e ci ho messo non poco tempo. Vedendo i vostri mi sembra abbiate utilizzato un api che dovrebbe oggettivamente semplificare il lavoro. 

l'api è pubblica?

**Comment by aborruso** [comment]
Ciao @giuspataro è pubblica, nel senso che ne vedi il funzionamento semplicemente leggendo il traffico web, con il browser:

![image](https://user-images.githubusercontent.com/30607/192892536-8dcbef04-5e9b-4f57-b2bf-54a6a76fd387.png)

E quella chiamata, leggendola da browser è

```
curl 'https://eleapi.interno.gov.it/siel/PX/votantiP/DE/20220925/TE/02' \
  -H 'authority: eleapi.interno.gov.it' \
  -H 'accept: application/json, text/javascript, */*; q=0.01' \
  -H 'accept-language: en-US,en;q=0.9,it;q=0.8' \
  -H 'content-type: application/json' \
  -H 'if-modified-since: Mon, 26 Sep 2022 15:08:10 GMT' \
  -H 'if-none-match: W/"0900fca8bb15e2fbaef705344fed4317"' \
  -H 'origin: https://elezioni.interno.gov.it' \
  -H 'referer: https://elezioni.interno.gov.it/' \
  -H 'sec-ch-ua: "Google Chrome";v="105", "Not)A;Brand";v="8", "Chromium";v="105"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Windows"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-site' \
  -H 'user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36' \
  --compressed
```

Non è pero documentata. Quindi abbiamo esplorato un po' il sito, letto il traffico e dedotto volta per volta la struttura..

**Comment by giuspataro** [comment]
ciao @aborruso dopo una piccola ricerca fatta ieri mi sono reso conto che effettivamente basta cercare l'api sul traffico.

In realtà mi avete aperto un mondo, perché solitamente usavo le classiche librerie per fare scraping (selenium, bs4 etc) e non avevo mai provato a tirare giù dati analizzando il flusso e utilizzando le api. 

Quindi come dire, oltre che fornire tantissimi dati, fate anche formazione!

Grazie ancora,

Giuseppe

**Reply by aborruso** [reply]
Fa piacere. Se produrrai qualcosa a valle di questa informazioni che hai appreso, e ce lo vorrai condividere, ci farà molto piacere
